pkg/subctl/cmd/gather/logs.go

```diff
@@ -97,20 +97,21 @@ func getLogFromStream(logStream io.ReadCloser) (string, error) {
 	return logs.String(), nil
 }
 
-func writeLogToFile(data, podName string, info Info, fileExtension string) error {
-	fileName := filepath.Join(info.DirName, escapeFileName(info.ClusterName+"_"+podName)+fileExtension)
-	f, err := os.Create(fileName)
+func writeLogToFile(data, podName string, info Info, fileExtension string) (*string, error) {
```

Comment:
There's really no reason to return a string pointer - if an error occurs return empty string. 

```suggestion
func writeLogToFile(data, podName string, info Info, fileExtension string) (string, error) {
```